### ISSUE TITLE
ObjectIntMap.get() causes ArrayIndexOutOfBoundsException in rare cases

### ISSUE THREAD

**Issue by JavaNoobsyBoobsy** [libgdx/libgdx#3826]
java.lang.ArrayIndexOutOfBoundsException: length=0; index=8
        at java.lang.String.equals(String.java:662)
        at com.badlogic.gdx.utils.ObjectIntMap.get(UniWar:281)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.fetchAttributeLocation(UniWar:287)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.disableVertexAttribute(UniWar:714)
        at com.badlogic.gdx.graphics.glutils.VertexArray.unbind(UniWar:157)
        at com.badlogic.gdx.graphics.Mesh.unbind(UniWar:398)
        at com.badlogic.gdx.graphics.Mesh.unbind(UniWar:389)
        at com.badlogic.gdx.graphics.Mesh.render(UniWar:501)
        at com.badlogic.gdx.graphics.Mesh.render(UniWar:449)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.flush(UniWar:971)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.switchTexture(UniWar:1050)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.draw(UniWar:364)

**Comment by xoppa** [comment]
https://github.com/libgdx/libgdx/wiki/More-Info

**Comment by davebaol** [comment]
@JavaNoobsyBoobsy 
Just to be sure, are you changing the map in multi-threading?
Unless otherwise stated in the javadoc, all libgdx collections are not thread-safe.

**Comment by JavaNoobsyBoobsy** [comment]
We are not modifying this map and not changing any mesh, nor any shader. That's why I reported the error as I imagine that it could be an implementation issue in the class ObjectIntMap itself. I am not sure what triggers the issue and that would be the interesting point to understand:

**Comment by JavaNoobsyBoobsy** [comment]
The code below reproduce the issue. (we don't use ObjectIntMap in our code). Anyway, I don't think that this issue is very important... You can close it if you want.
 
   public static void main(String[] args) {
        ObjectIntMap o = new ObjectIntMap(1);
        o.resize(0);
        o.get("A", 5);
}

**Comment by JavaNoobsyBoobsy** [comment]
This happened 2/6/2016 9:38:18 on Android device: Lenovo sisleyr	Lenovo S90-A, resolution:720x1280, OS:5.0.2. Here is the full stack trace for completeness and to eliminate the possibility that it is a multi-threaded issue (the app has a single thread design): "java.lang.ArrayIndexOutOfBoundsException: length=0; index=8
        at java.lang.String.equals(String.java:662)
        at com.badlogic.gdx.utils.ObjectIntMap.get(ObjectIntMap.java:281)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.fetchAttributeLocation(ShaderProgram.java:287)
        at com.badlogic.gdx.graphics.glutils.ShaderProgram.disableVertexAttribute(ShaderProgram.java:714)
        at com.badlogic.gdx.graphics.glutils.VertexArray.unbind(VertexArray.java:157)
        at com.badlogic.gdx.graphics.Mesh.unbind(Mesh.java:398)
        at com.badlogic.gdx.graphics.Mesh.unbind(Mesh.java:389)
        at com.badlogic.gdx.graphics.Mesh.render(Mesh.java:501)
        at com.badlogic.gdx.graphics.Mesh.render(Mesh.java:449)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.flush(SpriteBatch.java:971)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.switchTexture(SpriteBatch.java:1050)
        at com.badlogic.gdx.graphics.g2d.SpriteBatch.draw(SpriteBatch.java:364)
        at jg.Graphics.drawRegion(Graphics.java:289)
        at jg.Graphics.drawRegion(Graphics.java:243)
        at tbs.graphics.Gob.paint(Gob.java:511)
        at tbs.graphics.AnimSet.paintFrame(AnimSet.java:631)
        at tbs.graphics.AnimPlayer.paint(AnimPlayer.java:963)
        at tbs.graphics.AnimPlayer.paint(AnimPlayer.java:977)
        at uniwar.game.ui.GamePainter.paintAnim(GamePainter.java:127)
        at uniwar.game.ui.BaseAnimPlayers.paintBase(BaseAnimPlayers.java:187)
        at uniwar.game.ui.GamePainter.paintBase(GamePainter.java:414)
        at uniwar.game.ui.GamePainter.paintBases(GamePainter.java:400)
        at uniwar.game.ui.GamePainter.access$100(GamePainter.java:46)
        at uniwar.game.ui.GamePainter.access$400(GamePainter.java:46)
        at uniwar.game.ui.GamePainter.access$600(GamePainter.java:46)
        at uniwar.game.ui.GamePainter.access$700(GamePainter.java:46)
        at uniwar.game.ui.GamePainter$UnitsAndBasesAndCursorPainter.paintHex(GamePainter.java:473)
        at uniwar.game.ui.GamePainter.paintAllHexes(GamePainter.java:611)
        at uniwar.game.ui.GamePainter.paintBattleField(GamePainter.java:168)
        at uniwar.game.model.Game.paint(Game.java:736)
        at uniwar.scene.ingame.InGameScene.paintGame(InGameScene.java:94)
        at uniwar.scene.ingame.InGameScene.paint(InGameScene.java:106)
        at tbs.scene.Stage.draw(Stage.java:740)
        at tbs.scene.SceneCanvas.paintGameView(SceneCanvas.java:88)
        at uniwar.UniWarCanvas.paintGameView(UniWarCanvas.java:518)
        at jg.Canvas.render(Canvas.java:497)
        at uniwar.UniWarGame.render(UniWarGame.java:50)
        at com.badlogic.gdx.backends.android.AndroidGraphics.onDrawFrame(AndroidGraphics.java:472)
        at android.opengl.GLSurfaceView$GLThread.guardedRun(GLSurfaceView.java:1522)
        at android.opengl.GLSurfaceView$GLThread.run(GLSurfaceView.java:1239)"

**Comment by Darkyenus** [comment]
```
public static void main(String[] args) {
ObjectIntMap o = new ObjectIntMap(1);
o.resize(0);
o.get("A", 5);
}
```

How can you call `resize` when it is an internal `private` method? Also, I don't believe that 0 is a legal argument to `resize` even if you do manage to call it.

**Comment by Tom-Ski** [comment]
These are two different errors, one throws out of bounds at String#equals (the error from the stacktrace you supplied), the other is due to invalid resizing of the ObjectIntMap (your example code)

Looks like this is an Android issue, probably due to some charset encoding.

**Comment by JavaNoobsyBoobsy** [comment]
I was completely wrong. I apologize for wasting your time on this. I will be much more carefully in reporting errors next time. I called the private resize(0) method after I decompiled the class and added a main() method to it to see how that exception could be reproduced. I didn't even notice that the real problem came from String.equals(). That device specific error is probably not worth investigating.
